tests/test_commands.py

```diff
@@ -95,6 +95,43 @@ def test_client_setname(self, r):
         assert r.client_setname('redis_py_test')
         assert r.client_getname() == 'redis_py_test'
 
+    @skip_if_server_version_lt('2.4.0')
+    def test_client_kill(self, r, r3):
+        clients = r.client_list()
```

Comment:
An assert here to prove there are at least 4 connected test clients would help illustrate that the test is actually killing one of them.

Note that exact counts here will fail if additional non-test runner clients are connected to the Redis server. For example, if someone is connected to their local Redis server via redis-cli or a Python repl, that connection will be counted in the client list. We either need to filter those out so that we're only counting clients created by the test runner.

Reply:
Hey, this is quite tricky. This is what I've done:
```
      @skip_if_server_version_lt('2.4.0')
    def test_client_kill(self, r, r2):
        r.client_setname('c1')
        r2[0].client_setname('c2')
        r2[1].client_setname('c3')
        clients = r.client_list()
        own_clients = [client for client in clients 
                       if client.get('name') in ['c1', 'c2', 'c3']]
        assert len(own_clients) == 3
        resp = r.client_kill(clients[1].get('addr'))
        assert isinstance(resp, bool) and resp is True
        clients = r.client_list()
        own_clients = [client for client in clients 
                       if client.get('name') in ['c1', 'c2', 'c3']]
        assert len(own_clients) == 2
```

For some reason, I cannot use the r.client_id() as I get an error from redis and I have to somehow filter them out from the external ones.
What do you think?

Reply:
@theodesp what error? note that client_id is a v5 feature, so you need to have the proper version installed.

Reply:
It looks to me something is wrong in the server or something.
```
➜ uname -a
Darwin C02PV2X1FVH6 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64
➜ redis-server --version
Redis server v=5.0.2 sha=00000000:0 malloc=libc bits=64 build=25bb354a43384ae5
➜ ps aux | grep redis-server
itspare          70596   0.0  0.2  4335252  26340   ??  SN   Mon12pm   1:26.63 redis-server *:6379
➜ redis-cli
127.0.0.1:6379> client ID
(error) ERR Syntax error, try CLIENT (LIST | KILL | GETNAME | SETNAME | PAUSE | REPLY)

Reply:
Nope - your running redis-server is not-obviously older (I can tell from the error message). Do `INFO SERVER` from the the cli to see the real version, then `SHUTDOWN` and restart.

Reply:
A yes I had to restart the server as I did recently upgrade it. Thank you!

However, I would like  to avoid using r.client_id() in this test case as I will have to pump the min server requirements.

Reply:
Sure thing - was just interested in clarifying the error thingie.

Reply:
@theodesp You can use `client_setname` and `client_getname` to do the filtering. That was added in 2.6.9. I'd just set each client to have a known prefix like `'redis-py-test-...'`.

It looks like you're kind of already doing that in the code you copied above. But you need to use `r.client_getname` to read back the result.